### DISCUSSION TITLE
Installation process

### DISCUSSION THREAD

**Discussion by delchere** [nakib/elphbolt#117] [General]
Good morning Sir, please I have been trying to install elphbolt but running into issues.  while using Linux, I downloaded first of all the elphbolt, and then I run the dockerfile. it runs without error but I can't find any files that were execute. So, I am really don't know, maybe I am doing it in the wrong way.

**Comment by nakib** [comment]
Hi Delchere,

Thanks for writing. After installation, what material are you running `elphbolt` on? You need to generate some prerequisite DFT/Wannier data to be able to run this code. Have you taken a look at the silicon example? It gives all the inputs needed to generate the DFT/Wannier data. Alternatively, you can download the full set of data for silicon carbide by saying: `curl -X GET "https://nomad-lab.eu/prod/v1/api/v1/uploads/5b-c4JYRSG6qitMFjFUUzQ/raw/?offset=0&length=-1&compress=true" -o 3C-SiC.zip`

Or was your question more about using the Docker image of `elphbolt`? Out of curiosity, do you need to use Docker for your project?

Let me know if this helps.

Best,
Nakib

**Comment by delchere** [comment]
The real problem is the installation of elphbolt. i don't know how to
install it.  i have already saw the silicon example, i have even started
with it(phonon, epw).


On Sat, Oct 28, 2023 at 11:17 AM Nakib H. Protik ***@***.***>
wrote:

> Hi Delchere,
>
> Thanks for writing. After installation, what material are you running
> elphbolt on? You need to generate some prerequisite DFT/Wannier data to
> be able to run this code. Have you taken a look at the silicon example? It
> gives all the inputs needed to generate the DFT/Wannier data.
> Alternatively, you can download the full set of data for silicon carbide by
> saying: curl -X GET "
> https://nomad-lab.eu/prod/v1/api/v1/uploads/5b-c4JYRSG6qitMFjFUUzQ/raw/?offset=0&length=-1&compress=true"
> -o 3C-SiC.zip
>
> Or was your question more about using the Docker image of elphbolt? Out
> of curiosity, do you need to use Docker for your project?
>
> Let me know if this helps.
>
> Best
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nakib/elphbolt/discussions/117#discussioncomment-7410757>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BDSFO3PE6Y65MTKVILNKPE3YBTSVTAVCNFSM6AAAAAA6TAAIC6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TIMJQG42TO>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Delchere D0N-TSA
*Phd Student*|* ARISE-Project*

*Alumnus*| *AIMS-Cameroon*



*University of Lome *|* Togo*

*The Next Einstein Initiative* *|* *www.aims-cameroon.org*
<http://www.aims-cameroon.org/>

Mobile: +237 656160101 | +228 92069498

P.O. Box 1515, Klikamè , Lome, Togo

**Reply by nakib** [reply]
I see. At the present, one has to install the 2 dependencies, `spglib` and `opencoarrays` before building `elphbolt`. Have you tried following the instructions here: https://github.com/nakib/elphbolt#install-on-hpc-systems-with-gcc

Both the `spglib` and `opencoarrays` projects have documentation on how to build on various systems. You can install these on your HPC system by yourself or ask an admin to create the modules.

If, instead, you are on a Debian-based machine with root access, you can install some of the dependencies with 
```sudo apt install cmake mpich liblapack-dev libsymspg-dev```

You might also be able to install `opencoarrays` as shown above, though I am not sure if your OS software repository will have a new enough version. It might be worth trying it first. And if this doesn't work, you will have to build from source. You can use the provided Dockerfile as a guide to see how it can be built from source.

After you have done the above, you can build `elphbolt` with the `cmake` option.

If you really need a `docker` image, you can look up its documentation/tutorials regarding how to build an image from a Dockerfile and how to run it. I, personally, either use the `fpm` or `cmake` builds.

Let me know if this was useful for you.

Best regards,
Nakib

**Comment by delchere** [comment]
Thank you very much. I thought that process was only for the hpc. I will
try it

On Mon, Oct 30, 2023 at 9:23 AM Nakib H. Protik ***@***.***>
wrote:

> I see. At the present, one has to install the 2 dependencies, spglib and
> opencoarrays before building elphbolt. Have you tried following the
> instructions here:
> https://github.com/nakib/elphbolt#install-on-hpc-systems-with-gcc
>
> Both the spglib and opencoarrays projects have documentation on how to
> build on various systems. You can install these on your HPC system by
> yourself or ask an admin to create the modules.
>
> If, instead, you are on a Debian-based machine with root access, you can
> install some of the dependencies with
> sudo apt install cmake mpich liblapack-dev libsymspg-dev
>
> You might also be able to install opencoarrays as shown above, though I
> am not sure if your OS software repository will have a new enough version.
> It might be worth trying it first. And if this doesn't work, you will have
> to build from source. You can use the provided Dockerfile as a guide to see
> how it can be built from source.
>
> After you have done the above, you can build elphbolt with the cmake
> option.
>
> If you really need a docker image, you can look up its
> documentation/tutorials regarding how to build an image from a Dockerfile
> and how to run it. I, personally, either use the fpm or cmake builds.
>
> Let me know if this was useful for you.
>
> Best regards,
> Nakib
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nakib/elphbolt/discussions/117#discussioncomment-7421661>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BDSFO3K5AGSRSBZBWHX2FW3YB5W2RAVCNFSM6AAAAAA6TAAIC6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TIMRRGY3DC>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Delchere D0N-TSA
*Phd Student*|* ARISE-Project*

*Alumnus*| *AIMS-Cameroon*



*University of Lome *|* Togo*

*The Next Einstein Initiative* *|* *www.aims-cameroon.org*
<http://www.aims-cameroon.org/>

Mobile: +237 656160101 | +228 92069498

P.O. Box 1515, Klikamè , Lome, Togo

**Comment by delchere** [comment]
While cloning the elphbolt, I am getting the following error.


Cloning into 'elphbolt'...
remote: Enumerating objects: 3838, done.
remote: Counting objects: 100% (700/700), done.
remote: Compressing objects: 100% (234/234), done.
remote: Total 3838 (delta 499), reused 656 (delta 461), pack-reused 3138
Receiving objects: 100% (3838/3838), 3.76 MiB | 4.56 MiB/s, done.
Resolving deltas: 100% (2879/2879), done.
-- The Fortran compiler identification is GNU 11.4.0
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Check for working Fortran compiler: /usr/bin/f95 - skipped
CMake Error at CMakeLists.txt:26 (find_library):
  Could not find SPG_LIB using the following names: symspg


-- Configuring incomplete, errors occurred!
See also "/home/lqa-arise-del/elphbolt/build/CMakeFiles/CMakeOutput.log".
make: *** No targets specified and no makefile found.  Stop.
***@***.***:~$

On Mon, Oct 30, 2023 at 5:36 AM Delchere Dontsa <
***@***.***> wrote:

> Thank you very much. I thought that process was only for the hpc. I will
> try it
>
> On Mon, Oct 30, 2023 at 9:23 AM Nakib H. Protik ***@***.***>
> wrote:
>
>> I see. At the present, one has to install the 2 dependencies, spglib and
>> opencoarrays before building elphbolt. Have you tried following the
>> instructions here:
>> https://github.com/nakib/elphbolt#install-on-hpc-systems-with-gcc
>>
>> Both the spglib and opencoarrays projects have documentation on how to
>> build on various systems. You can install these on your HPC system by
>> yourself or ask an admin to create the modules.
>>
>> If, instead, you are on a Debian-based machine with root access, you can
>> install some of the dependencies with
>> sudo apt install cmake mpich liblapack-dev libsymspg-dev
>>
>> You might also be able to install opencoarrays as shown above, though I
>> am not sure if your OS software repository will have a new enough version.
>> It might be worth trying it first. And if this doesn't work, you will have
>> to build from source. You can use the provided Dockerfile as a guide to see
>> how it can be built from source.
>>
>> After you have done the above, you can build elphbolt with the cmake
>> option.
>>
>> If you really need a docker image, you can look up its
>> documentation/tutorials regarding how to build an image from a Dockerfile
>> and how to run it. I, personally, either use the fpm or cmake builds.
>>
>> Let me know if this was useful for you.
>>
>> Best regards,
>> Nakib
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/nakib/elphbolt/discussions/117#discussioncomment-7421661>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/BDSFO3K5AGSRSBZBWHX2FW3YB5W2RAVCNFSM6AAAAAA6TAAIC6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TIMRRGY3DC>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>
>
> --
> Delchere D0N-TSA
> *Phd Student*|* ARISE-Project*
>
> *Alumnus*| *AIMS-Cameroon*
>
>
>
> *University of Lome *|* Togo*
>
> *The Next Einstein Initiative* *|* *www.aims-cameroon.org*
> <http://www.aims-cameroon.org/>
>
> Mobile: +237 656160101 | +228 92069498
>
> P.O. Box 1515, Klikamè , Lome, Togo
>


-- 
Delchere D0N-TSA
*Phd Student*|* ARISE-Project*

*Alumnus*| *AIMS-Cameroon*



*University of Lome *|* Togo*

*The Next Einstein Initiative* *|* *www.aims-cameroon.org*
<http://www.aims-cameroon.org/>

Mobile: +237 656160101 | +228 92069498

P.O. Box 1515, Klikamè , Lome, Togo

**Reply by nakib** [reply]
You have to include the header file and link the shared library of `spglib`. See more here: https://spglib.readthedocs.io/en/latest/install.html 

You can do this once and for all by adding the following to your `~/.bashrc` file:

```
export PATH="<path to spglib build directory>:$PATH"
export LD_LIBRARY_PATH="<path to spglib build directory>:$LD_LIBRARY_PATH"
```

**Comment by delchere** [comment]
and while building with fpm, i have the
following
(base) ***@***.***:~/elphbolt$ conda
deactivate
***@***.***:~/elphbolt$ .
fpm_config_caf.sh; yes | fpm clean; fpm build; fpm test
Missing required -s flag. What package source did you want? {:level=>:warn}
Missing required -t flag. What package output did you want? {:level=>:warn}
Fix the above problems, and you'll be rolling packages in no time!
{:level=>:fatal}


On Mon, Oct 30, 2023 at 10:56 AM Delchere Dontsa <
***@***.***> wrote:

> While cloning the elphbolt, I am getting the following error.
>
>
> Cloning into 'elphbolt'...
> remote: Enumerating objects: 3838, done.
> remote: Counting objects: 100% (700/700), done.
> remote: Compressing objects: 100% (234/234), done.
> remote: Total 3838 (delta 499), reused 656 (delta 461), pack-reused 3138
> Receiving objects: 100% (3838/3838), 3.76 MiB | 4.56 MiB/s, done.
> Resolving deltas: 100% (2879/2879), done.
> -- The Fortran compiler identification is GNU 11.4.0
> -- Detecting Fortran compiler ABI info
> -- Detecting Fortran compiler ABI info - done
> -- Check for working Fortran compiler: /usr/bin/f95 - skipped
> CMake Error at CMakeLists.txt:26 (find_library):
>   Could not find SPG_LIB using the following names: symspg
>
>
> -- Configuring incomplete, errors occurred!
> See also "/home/lqa-arise-del/elphbolt/build/CMakeFiles/CMakeOutput.log".
> make: *** No targets specified and no makefile found.  Stop.
> ***@***.***:~$
>
> On Mon, Oct 30, 2023 at 5:36 AM Delchere Dontsa <
> ***@***.***> wrote:
>
>> Thank you very much. I thought that process was only for the hpc. I will
>> try it
>>
>> On Mon, Oct 30, 2023 at 9:23 AM Nakib H. Protik ***@***.***>
>> wrote:
>>
>>> I see. At the present, one has to install the 2 dependencies, spglib
>>> and opencoarrays before building elphbolt. Have you tried following the
>>> instructions here:
>>> https://github.com/nakib/elphbolt#install-on-hpc-systems-with-gcc
>>>
>>> Both the spglib and opencoarrays projects have documentation on how to
>>> build on various systems. You can install these on your HPC system by
>>> yourself or ask an admin to create the modules.
>>>
>>> If, instead, you are on a Debian-based machine with root access, you can
>>> install some of the dependencies with
>>> sudo apt install cmake mpich liblapack-dev libsymspg-dev
>>>
>>> You might also be able to install opencoarrays as shown above, though I
>>> am not sure if your OS software repository will have a new enough version.
>>> It might be worth trying it first. And if this doesn't work, you will have
>>> to build from source. You can use the provided Dockerfile as a guide to see
>>> how it can be built from source.
>>>
>>> After you have done the above, you can build elphbolt with the cmake
>>> option.
>>>
>>> If you really need a docker image, you can look up its
>>> documentation/tutorials regarding how to build an image from a Dockerfile
>>> and how to run it. I, personally, either use the fpm or cmake builds.
>>>
>>> Let me know if this was useful for you.
>>>
>>> Best regards,
>>> Nakib
>>>
>>> —
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/nakib/elphbolt/discussions/117#discussioncomment-7421661>,
>>> or unsubscribe
>>> <https://github.com/notifications/unsubscribe-auth/BDSFO3K5AGSRSBZBWHX2FW3YB5W2RAVCNFSM6AAAAAA6TAAIC6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TIMRRGY3DC>
>>> .
>>> You are receiving this because you authored the thread.Message ID:
>>> ***@***.***>
>>>
>>
>>
>> --
>> Delchere D0N-TSA
>> *Phd Student*|* ARISE-Project*
>>
>> *Alumnus*| *AIMS-Cameroon*
>>
>>
>>
>> *University of Lome *|* Togo*
>>
>> *The Next Einstein Initiative* *|* *www.aims-cameroon.org*
>> <http://www.aims-cameroon.org/>
>>
>> Mobile: +237 656160101 | +228 92069498
>>
>> P.O. Box 1515, Klikamè , Lome, Togo
>>
>
>
> --
> Delchere D0N-TSA
> *Phd Student*|* ARISE-Project*
>
> *Alumnus*| *AIMS-Cameroon*
>
>
>
> *University of Lome *|* Togo*
>
> *The Next Einstein Initiative* *|* *www.aims-cameroon.org*
> <http://www.aims-cameroon.org/>
>
> Mobile: +237 656160101 | +228 92069498
>
> P.O. Box 1515, Klikamè , Lome, Togo
>


-- 
Delchere D0N-TSA
*Phd Student*|* ARISE-Project*

*Alumnus*| *AIMS-Cameroon*



*University of Lome *|* Togo*

*The Next Einstein Initiative* *|* *www.aims-cameroon.org*
<http://www.aims-cameroon.org/>

Mobile: +237 656160101 | +228 92069498

P.O. Box 1515, Klikamè , Lome, Togo

**Reply by nakib** [reply]
Hmm, I do not use `conda`, so I can't really comment on this. But you seem to be missing the `. ` before `fpm_config_caf.sh`. The symbol `.` is a short-hand for the unix command `source`.
